operations/src/main/java/edu/unc/lib/boxc/operations/impl/versioning/VersionedDatastreamService.java

```diff
@@ -71,6 +73,14 @@ public BinaryObject addVersion(DatastreamVersion newVersion) {
 
         Lock dsLock = lockManager.awaitWriteLock(dsPid);
         BinaryObject dsObj = getBinaryObject(dsPid);
+        if (dsObj != null && newVersion.getUnmodifiedSince() != null) {
+            Date lastModified = dsObj.getLastModified();
+            if (dsObj.getLastModified().toInstant().isAfter(newVersion.getUnmodifiedSince())) {
```

Comment:
maybe using `isBefore` so that the `getUnmodifiedSince` comes first and the `lastModified` comes after, which reflects the actual logic, would be better? along with flipping around the error message

Reply:
maybe update the variable to `fedoraLastModified = dsObj.getLastModified().toInstant()` and then in the if statement you can do 
`if newVersion.getUnmodifiedSince().isBefore(fedoraLastModified)`
which would make it a bit easier to understand. and then you can use the same variable